File: src/java/foundgine-core/src/main/java/com/foundgine/core/semantic/SemanticRelationship.java
package com.foundgine.core.semantic;
import com.foundgine.core.abstractions.*;
import java.util.*;
public record SemanticRelationship(RelationshipId id, String name, EntityId target, RelationshipCardinality cardinality, List<SemanticAlias> aliases) {
    public SemanticRelationship { aliases = aliases == null ? List.of() : List.copyOf(aliases); }
    public SemanticRelationship(RelationshipId id, String name, EntityId target, RelationshipCardinality cardinality) { this(id,name,target,cardinality,List.of()); }
    public List<SemanticAlias> effectiveAliases() { return aliases; }
}

File: src/csharp/Foundgine.Core/Semantic/SemanticRelationship.cs
using Foundgine.Core.Abstractions;

namespace Foundgine.Core.Semantic;

/// <summary>
/// A named semantic relationship between two entities.
/// </summary>
public sealed record SemanticRelationship(
    RelationshipId Id,
    string Name,
    EntityId Target,
    RelationshipCardinality Cardinality,
    IReadOnlyList<SemanticAlias>? Aliases = null)
{
    public IReadOnlyList<SemanticAlias> EffectiveAliases => Aliases ?? [];
}
